### ISSUE TITLE
Make party fields in the docket search results to maintain order consistency

### ISSUE THREAD

**Issue by albertisfu** [freelawproject/courtlistener#4531]
In the following query: https://www.courtlistener.com/api/rest/v4/search/?q=antitrust&type=r

You can observe that the first `attorney` is listed as `James F Keller`, but the first `attorney_id` is `269313`, which belongs to a different attorney. As a result, the `attorney` and `attorney_id` lists do not match due to the differing order.

This issue also occurs with other party related fields such as `party`, `party_id`, `firm`, and `firm_id`.

The problem is that we're using a `set()` to build these lists. Since sets in Python are unordered, the order of values added to the set is not guaranteed.

To resolve this, we should ensure that the lists of parties maintain the correct order. Using a `list()` instead of a `set()` may be sufficient, as long as we take care to prevent duplicates when indexing parties.

**Comment by mlissner** [comment]
Thanks Alberto. Seems simple enough to fix going forward, but I assume the real fix will require a full re-index. So it goes!

**Comment by albertisfu** [comment]
Yeah that's correct this fix would require a full-reindex but only for Dockets. 

We already have a reindex of dockets planned for a different issue related to parties:

https://github.com/freelawproject/infrastructure/issues/157

So perhaps that re-index can wait a bit and address this issue at the same time?
